### DISCUSSION TITLE
How to delete all data in repeater ?

### DISCUSSION THREAD

**Discussion by jigujigu** [filamentphp/filament#4543] [Help] [answered]
Hello,

How I can check is the value is changed or not in Select().
And if changed, How I can delete one or all of detail data  in repeater using afterStateUpdated() ?

Thank You.

```
Select::make('vendor_id')
->label('Vendor')
->relationship('vendor', 'name')
->reactive()
->afterStateUpdated(function (Closure $state, Closure $set) {
        // How I can delete all my Repeater if vendor_id is changed ??

    }),


Repeater::make('invoicedetails')
->relationship('invoicedetails')
->schema([
    Select::make('orderdetail_id')
     xxxx
     xxxx
     xxxx
```

**Comment by danharrin** [comment, marked as answer]
Probably `$set('invoicedetails', [])`

**Reply by jigujigu** [reply]
Hi @danharrin ,

The repeater become empty after I do `$set('invoicedetails',[])`, before that there were 3 lines.
![image](https://user-images.githubusercontent.com/6214964/196047712-51984585-df63-4965-8010-cd967f4f0682.png)

But, when I click Save Button,. 
The detail is show-up again with empty value.

![image](https://user-images.githubusercontent.com/6214964/196048237-8833fd3f-1180-4108-bfca-bc16725b11d7.png)


I need advice again.
Thank  You

**Comment by jigujigu** [comment]
Hi danharrin,
Thank you for your advice.
